ckanext/stadtzhharvest/harvester.py

```diff
@@ -367,6 +378,18 @@ def _import_package(self, harvest_object):
         Session.commit()
         return True
 
+    def _delete_dataset(self, package_dict):
+        try:
```

Comment:
I didn't realize earlier, but the try/except is actually not needed, since we already catch exceptions on a higher level: https://github.com/opendatazurich/ckanext-stadtzh-harvest/blob/2e51a89f421443c8d3e3d294780ca71571af152f/ckanext/stadtzhharvest/harvester.py#L270-L283

So simply call dataset_purge and return True, if dataset purge fails, the except block above will catch it and return False.